src/arrayMethodJoin.js

```diff
@@ -5,7 +5,20 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    const sourceStr = [...this].toString();
+    let result = '';
+
+    for (let i = 0; i < sourceStr.length; i++) {
+      if (sourceStr[i] === ',') {
+        result += separator;
+      } else if (separator === undefined) {
+        return sourceStr;
+      } else {
+        result += sourceStr[i];
+      }
+    }
```

Comment:
Do not use else statement

```suggestion
    for (let i = 0; i < this.length; i++) {
    
      // Add each item to result,

       // if item is not last add separator
    }
```